variant.d

```diff
@@ -234,6 +234,12 @@ struct Variant{
 		assert(occupies == Occupies.vars);
 	}
 
+	this()(typeof(null), Type type = null){ // templated because of DMD bug
```

Comment:
I wasn't super confident about this change: is it the expected ctor if the first argumet is null?